test_files/rest_parameters_generic_empty/rest_parameters_generic_empty.js

```diff
@@ -0,0 +1,24 @@
+// test_files/rest_parameters_generic_empty/rest_parameters_generic_empty.ts(3,1): warning TS0: var args type is not an object type
+// test_files/rest_parameters_generic_empty/rest_parameters_generic_empty.ts(3,1): warning TS0: var args type is not an object type
+/**
+ * @fileoverview added by tsickle
+ * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
+ */
+goog.module('test_files.rest_parameters_generic_empty.rest_parameters_generic_empty');
+var module = module || { id: 'test_files/rest_parameters_generic_empty/rest_parameters_generic_empty.ts' };
+module = module;
+exports = {};
+/**
+ * @template A
+ * @param {function(!Array<?>): void} fn
```

Comment:
We inline the lower bound? Then why emit `@template A` at all?

Reply:
We unconditionally emit `@template A`. We don't really inline the bound, we just fail to handle rest arguments that use template type variables from the outer scope.